src-tauri/src/main.rs

```diff
@@ -108,5 +131,120 @@ fn main() {
       api.prevent_exit();
     }
     _ => {}
+  });
+
+  Ok(())
+}
+
+/// Start farming by using plot in specified path and connecting to WebSocket server at specified
+/// address.
+pub(crate) async fn farm(
+  base_directory: PathBuf,
+  node_rpc_url: &str,
+  ws_server_listen_addr: SocketAddr,
+) -> Result<[u8; 32], anyhow::Error> {
+  // TODO: This doesn't account for the fact that node can
+  // have a completely different history to what farmer expects
+  info!("Opening plot");
+  let plot_fut = tokio::task::spawn_blocking({
+    let base_directory = base_directory.clone();
+
+    move || Plot::open_or_create(&base_directory)
+  });
+  let plot = plot_fut.await.unwrap()?;
+
+  info!("Opening commitments");
+  let commitments_fut = tokio::task::spawn_blocking({
+    let path = base_directory.join("commitments");
+
+    move || Commitments::new(path)
+  });
+  let commitments = commitments_fut.await.unwrap()?;
+
+  info!("Opening object mapping");
+  let object_mappings = tokio::task::spawn_blocking({
+    let base_directory = base_directory.clone();
+
+    move || ObjectMappings::open_or_create(&base_directory)
   })
+  .await??;
+
+  info!("Connecting to node at {}", node_rpc_url);
+  let client = WsRpc::new(node_rpc_url).await?;
+
+  let farmer_metadata = client
+    .farmer_metadata()
+    .await
+    .map_err(|error| anyhow::Error::msg(error.to_string()))?;
+
+  let identity = Identity::open_or_create(&base_directory)?;
+
+  let subspace_codec = SubspaceCodec::new(identity.public_key());
+
+  // Start RPC server
+  let ws_server = WsServerBuilder::default()
+    .build(ws_server_listen_addr)
+    .await?;
+  let ws_server_addr = ws_server.local_addr()?;
+  let rpc_server = RpcServerImpl::new(
+    farmer_metadata.record_size,
+    farmer_metadata.recorded_history_segment_size,
+    plot.clone(),
+    object_mappings.clone(),
+    subspace_codec,
+  );
+  let _stop_handle = ws_server.start(rpc_server.into_rpc())?;
+
+  info!("WS RPC server listening on {}", ws_server_addr);
```

Comment:
Not sure we need a WebSocket server, maybe not by default (can be added later)

Reply:
I'm not getting this. Isn't this required? Without these lines, I'm getting this error:
```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Networking or low-level protocol error: Error when opening the TCP socket: Connection refused (os error 61)

Caused by:
    Error when opening the TCP socket: Connection refused (os error 61)', src/main.rs:36:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Networking or low-level protocol error: Error when opening the TCP socket: Connection refused (os error 61)
```